mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/MavenProjectConfigurator.java

```diff
@@ -70,10 +70,9 @@ private void addSkipAfterFailureCount(PluginExecution pluginExecution) {
         final Xpp3Dom count = pluginConfiguration.getChild(skipAfterFailureCountProperty);
         if (count == null) {
             final Xpp3Dom xpp3Dom = new Xpp3Dom(skipAfterFailureCountProperty);
-            final String skipAfterFailureCount = System.getProperty("surefire.skipAfterFailureCount");
-            final String skipAfterFailureCountValue = skipAfterFailureCount != null ? skipAfterFailureCount : "1";
-
-            xpp3Dom.setValue(skipAfterFailureCountValue);
+            final String failureCount = System.getProperty("surefire.skipAfterFailureCount", "1");
```

Comment:
In case of `selecting` mode. I think user always want to execute all `new/changed/affected` tests, as anyways he has to verify behavior for  `new/changed/affected` tests. so at the end of run he can identify only failed tests from entire test set. 

e.g. If we enable `skipAfterFailureCount`  option only for `selecting` mode then execution will exit after single failure. But in this case user has to fix tests for that single failure & run it again. It might be possible in this case he will get next failure in second run, then again he has to fix & run it again.

Instead of this we should execute all tests. so he can get status for entire `new/changed/affected` test sets. So he can fix only failed tests & run build again.

User can opt-out using `surefire.skipAfterFailureCount` to `0` in case of any mode. In case of `failsafe` I will change it not to apply this flag.